rust/agama-server/src/users/web.rs

```diff
@@ -0,0 +1,229 @@
+//!
+//! The module offers two public functions:
+//!
+//! * `users_service` which returns the Axum service.
+//! * `users_stream` which offers an stream that emits the users events coming from D-Bus.
+
+use crate::{
+    error::Error,
+    web::{
+        common::{service_status_router, validation_router},
+        Event,
+    },
+};
+use agama_lib::{
+    error::ServiceError,
+    users::{proxies::Users1Proxy, FirstUser, UsersClient},
+};
+use axum::{extract::State, routing::get, Json, Router};
+use serde::{Deserialize, Serialize};
+use std::pin::Pin;
+use tokio_stream::{Stream, StreamExt};
+
+#[derive(Clone)]
+struct UsersState<'a> {
+    users: UsersClient<'a>,
+}
+
+/// Returns streams that emits users related events coming from D-Bus.
+///
+/// It emits the Event::RootPasswordChange, Event::RootSSHKeyChanged and Event::FirstUserChanged events.
+///
+/// * `connection`: D-Bus connection to listen for events.
+pub async fn users_streams(
+    dbus: zbus::Connection,
+) -> Result<Vec<(&'static str, Pin<Box<dyn Stream<Item = Event> + Send>>)>, Error> {
+    const FIRST_USER_ID: &str = "first_user";
+    const ROOT_PASSWORD_ID: &str = "root_password";
+    const ROOT_SSHKEY_ID: &str = "root_sshkey";
+    // here we have three streams, but only two events. Reason is
+    // that we have three streams from dbus about property change
+    // and munify two root user properties into single event to http API
```

Comment:
I am not sure whether we need a long explanation. Anyway, let's fix the typo.

```suggestion
    // and unify two root user properties into single event to http API
```